scalesetlistener/scalesetlistener.go

```diff
@@ -0,0 +1,368 @@
+package scalesetlistener
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/actions-runner-controller/actions-runner-controller/github"
+	"github.com/actions-runner-controller/actions-runner-controller/pkg/github/scalesetclient"
+	"github.com/go-logr/logr"
+)
+
+type Message struct {
+	// N is number of acquired jobs
+	N int
+}
+
+type Listener struct {
+	ghConfig *github.Config
+	config   *Config
+	logger   logr.Logger
+	message  chan *Message
+
+	builder *builder
+}
+
+type Config struct {
+	RunnerScaleSetName string
+	RunnerEnterprise   string
+	RunnerOrg          string
+	RunnerRepository   string
+}
+
+func New(config *Config, ghConfig *github.Config, logger logr.Logger, message chan *Message) *Listener {
+	return &Listener{
+		ghConfig: ghConfig,
+		config:   config,
+		logger:   logger,
+		message:  message,
+	}
+}
+
+func (l *Listener) MakeClients(ctx context.Context) error {
```

Comment:
I wanted to preserve go way of New making and initializing fields, but without calls. It is kind of implementation of a built-in `new(type)`. Because of that, I named Make clients to resemble `make()` built-in call. But thinking about your comment, `MakeListener` makes more sense, doesn't it?

Reply:
`MakeListener` is fine.

I think this is not an issue if you go DI route. We can move that logic outside of the implementation of MakeListener